caps/eth.md

```diff
@@ -108,22 +108,173 @@ relayed by the peer.
 Transactions must be validated before re-propagating them. Relaying an invalid transaction
 results in peer disconnection.
 
+### Transaction Encoding and Validity
+
+Transaction objects exchanged by peers have one of two encodings, which are identical to
+the encodings used by Ethereum consensus. In definitions across this specification, we
+refer to transaction objects of either encoding using the identifier `txₙ`.
+
+    tx = {legacy-tx, typed-tx}
+
+Untyped, legacy transactions are given as an RLP list.
+
+    legacy-tx = [
+        nonce: P,
+        gas-price: P,
+        gas-limit: P,
+        recipient: {B_0, B_20},
+        value: P,
+        data: B,
+        V: P,
+        R: P,
+        S: P,
+    ]
+
+[EIP-2718] typed transactions are encoded as RLP byte arrays where the first byte is the
+transaction type (`tx-type`) and the remaining bytes are opaque type-specific data.
+
+    typed-tx = rlp(tx-type || tx-data)
+
+Transactions must be validated when they are received. Validity depends on the Ethereum
+chain state. While the encoding of typed transactions is opaque, it is assumed that the
+encoding of `tx-data` provides values for `nonce`, `gas-price`, `gas-limit`, and that the
+sender account of the transaction can be determined from the signature.
+
+Transactions must be validated according to these rules:
+
+- If the transaction is typed, the `tx-type` must be one of the types considered
+  acceptable at the current block height.
+- The signature (`V`, `R`, `S` values) must be valid according to the signature schemes
+  supported by the chain. At this time, the two schemes in active use are the basic
+  'Homestead' scheme where `V`, `R`, `S` are simply the components of a secp256k1
+  signature; and the [EIP-155] scheme, where the signature values include a 'chain ID' to
+  prevent replay of the transaction on a different chain.
+- The `gas-limit` must cover the 'intrinsic gas' of the transaction, i.e. the base fee and
+  fee for the data size.
```

Comment:
As above, I recommend not including this in the networking protocol.  Each transaction type will have its own mechanisms for gas limits, and in theory some may not even include a gas limit (like the previously proposed cancellation transactions).